src/circuit_proof/assignment.rs

```diff
@@ -0,0 +1,135 @@
+use curve25519_dalek::scalar::Scalar;
+use errors::R1CSError;
+use std::ops::{Add, Div, Mul, Sub, Try};
+
+// The assignment value to a variable, as stored in `ConstraintSystem`.
+// Provers create a `Value` assignment, while verifiers create an `Missing` assignment.
+#[derive(Copy, Clone, Debug)]
+pub enum Assignment {
+    Value(Scalar),
+    Missing(),
+}
+
+impl Assignment {
+    pub fn new(scalar: Scalar) -> Self {
+        Assignment::Value(scalar)
+    }
+
+    pub fn zero() -> Self {
+        Assignment::Value(Scalar::zero())
+    }
+
+    pub fn from_u64(v: u64) -> Self {
```

Comment:
This seems like it could also be a `From` impl -- I guess it probably makes sense to provide multiple From impls instead of making users do two `from`s in a row?